subprojects/tooling-api/src/crossVersionTest/groovy/org/gradle/integtests/tooling/r56/TestLauncherDebugTestsCrossVersionTest.groovy

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright 2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.integtests.tooling.r56
+
+import com.sun.jdi.Bootstrap
+import com.sun.jdi.VirtualMachine
+import com.sun.jdi.connect.Connector
+import com.sun.jdi.connect.ListeningConnector
+import com.sun.jdi.connect.TransportTimeoutException
+import org.gradle.integtests.tooling.fixture.TargetGradleVersion
+import org.gradle.integtests.tooling.fixture.ToolingApiSpecification
+import org.gradle.integtests.tooling.fixture.ToolingApiVersion
+import org.gradle.tooling.BuildException
+import spock.lang.Timeout
+
+
+@ToolingApiVersion(">=5.6")
+@TargetGradleVersion(">=5.6")
+@Timeout(60)
+class TestLauncherDebugTestsCrossVersionTest extends ToolingApiSpecification {
```

Comment:
Do we need a test to explicitly test older TAPI clients with a newer Gradle?

Reply:
I don't think so. The existing TAPI cross-version tests for running tests should cover us I think.

Reply:
Resolving the conversation.